### PR TITLE
[SYCL] Add SYCL 2020 buffer and vec function byte_size()

### PR DESCRIPTION
**Pull request by nawrinsu**

### CODE REVIEW

File: sycl/include/CL/sycl/types.hpp [hunk 1]
```diff
@@ -1400,7 +1403,10 @@ class SwizzleOp {
   __SYCL2020_DEPRECATED("get_count() is deprecated, please use size() instead")
   size_t get_count() const { return size(); }
   size_t size() const noexcept { return getNumElements(); }
-  template <int Num = getNumElements()> size_t get_size() const {
+  __SYCL2020_DEPRECATED(
+      "get_size() is deprecated, please use byte_size() instead")
+  size_t get_size() const { return byte_size(); }
```
Comment (bader) [inline sycl/include/CL/sycl/types.hpp:1408]: I think `get_size` must be a template to avoid potential ABI breaks. Right?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
Could you please add tests for new functions?
Also you can add checks for deprecation messages in sycl/test/warnings/sycl_2020_deprecations.cpp.

**Comment by nawrinsu** [comment]
PR link for test - https://github.com/intel/llvm-test-suite/pull/403

**Review by vladimirlaz — APPROVED** [review]

**Comment by vladimirlaz** [comment]
@turinevgeny could you please review as the code owner?

**Review by turinevgeny — APPROVED** [review]

**Review by bader — APPROVED** [review]

**Pull request merged**